### PR TITLE
Consolidating Chrome + Firefox code

### PR DESCRIPTION
**Pull request by karllhughes**
Addresses #23 

In order to avoid duplicate code, I configured [Webpack](https://webpack.js.org/) to copy code into two folders:

- `/dist/chrome`
- `/dist/firefox`

I also added some instructions to the readme and put this under the MIT license. If you would rather pick another license, that's fine, but since Pokemon Showdown uses MIT, I figured it was safe for this project too.

Finally, I fixed the minor differences between the Chrome and Firefox codebases and tested them.

I realize this is a big change, but maintaining two versions of the plugin that are almost identical will make development _really_ tough in the future, and I'd like to continue developing this plugin.

Please let me know if you have feedback or if we need to change the Webpack config at all. Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,28 @@ Optional settings (right click the extension icon and toggle the setting!):
...
+- `firefox` - Firefox-specific Javascript and manifest.json files
+
+To build and test this code locally:
+
+```bash
+npm install
+npm run build
+```
+
+This will create a `dist/` folder with a `chrome/` and `firefox/` folder. Use these unpacked files as a Chrome or Firefox plugin to test locally.
```
Comment (karllhughes) [inline README.md:45]: Here's the instructions and an explanation of each of the folders/files.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "scripts": {
+    "build": "webpack"
+  },
+  "devDependencies": {
+    "clean-webpack-plugin": "^3.0.0",
+    "copy-webpack-plugin": "^6.0.3",
+    "webpack": "^4.43.0",
+    "webpack-cli": "^3.3.12"
+  }
+}
```
Comment (karllhughes) [inline package.json:14]: Webpack seems like the most popular way to do this sort of build process. If you'd prefer Grunt or Gulp, we could do that, but I have worked with Webpack most.

File: src/firefox/manifest.json [hunk 3]
```diff
@@ -1,5 +1,4 @@
 {
-	"update_url": "https://clients2.google.com/service/update2/crx",
```
Comment (karllhughes) [inline src/firefox/manifest.json:2]: This was causing an error in Firefox, so I removed it. Not sure if it needs to be a different URL or if it's just a copy/paste error from the Chrome version.

File: package.json (hunk 2 above)
Comment (rowin1) [reply package.json:14]: I prefer webpack, too and I think there are advantages to choosing the most popular way to do the build process 👍

File: README.md [hunk 4]
```diff
@@ -24,7 +24,28 @@ Optional settings (right click the extension icon and toggle the setting!):
...
+Bug reports and pull requests are welcome!  If you'd like to request a feature, please open an issue.  This project is intended to be a safe, welcoming space for collaboration.
+
+### Local Development
+The code for this plugin can be found in `./src`.
+
+- `index.js` - The primary codebase for this plugin
+- `icons` - Icon image files
+- `css` - Custom styles
+- `chrome` - Chrome-specific Javascript and manifest.json files
+- `firefox` - Firefox-specific Javascript and manifest.json files
```
Comment (rowin1) [inline README.md:36]: 🎉 🎉

File: src/firefox/manifest.json (hunk 3 above)
Comment (rowin1) [reply src/firefox/manifest.json:2]: Likely a copy/paste mistake. 👍

**Review by rowin1 — APPROVED** [review]

**Comment by ciesling** [comment]
Hi, I am sorry if this isn't the appropriate place to comment, but I have been using this chrome extension for over a year without many problems. Last Summer, I left a review on the Chrome Extension page requesting an update. It didn't take long for the developer to update the plugin and I was very grateful for the fix! Recently, I have been experiencing similar bugs on showdown with this plugin. For one, I am not able to view my pokemon's move sets and stats when hovering over my team while using this plugin. If It is possible to get in touch with the developer any other way so that he can see this, please let me know. Thank you so much for this plugin as I use it very often!

**Pull request merged**